sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -212,15 +220,27 @@ def get_all_items(self) -> List[dataset_item.DatasetItem]:
                     source=item_content.get("source"),  # type: ignore
                 )
 
-                stream_results.append(item)
+                results.append(item)
 
-            if len(stream_results) == 0:
-                break
```

Comment:
I’m afraid that without this check, the loop might never end if the streamer returns an empty string and the number of returned values is less than specified in the parameter nb_samples.